src/components/shared/table/Table.tsx

```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+import {
+  useFlexLayout,
+  usePagination,
+  useSortBy,
+  useTable,
+} from 'react-table';
+import { RiArrowLeftSLine, RiArrowRightSLine } from 'react-icons/ri';
+import PaginationButton from '../../../shared/buttons/PaginationButton';
+import { TableInstanceWithHooks, TableProps } from './types';
+
+function Table({ columns, data, children }: TableProps) {
+  const {
+    getTableProps,
+    getTableBodyProps,
+    headerGroups,
+    rows,
+    page,
+    prepareRow,
+    nextPage,
+    previousPage,
+    setPageSize,
+    canPreviousPage,
+    pageOptions,
+    canNextPage,
+    state: { pageIndex, pageSize },
+  } = useTable(
+    {
+      columns,
+      data,
+    },
+    useFlexLayout,
+    useSortBy,
+    usePagination,
+  ) as TableInstanceWithHooks<Object>;
+
+  const firstRowNumberOnPage = React.useMemo(() => {
+    const result = (pageIndex * pageSize) + 1;
+    return result;
+  }, [pageIndex, pageSize]);
+
+  const lastRowNumberOnPage = React.useMemo(() => {
+    const result = (pageIndex * pageSize) + (page.length);
+    return result;
+  }, [pageIndex, pageSize, page]);
+
+  return (
+    <div className="w-full h-full">
+      <div className="flex flex-col m-auto bg-transit-white">
+        <div className="flex flex-row justify-between items-center w-content px-4 py-2">
+          {children}
+        </div>
+        {/* eslint-disable-next-line react/jsx-props-no-spreading */}
+        <table {...getTableProps()}>
+          <thead className="bg-transit-grey">
+            {headerGroups.map((headerGroup) => {
+              const { key, ...restHeaderGroupProps } = headerGroup.getHeaderGroupProps();
+              return (
+              // eslint-disable-next-line react/jsx-props-no-spreading
```

Comment:
We need to look into this one to either disable on airbn config or keep like this. @KarolinaKopacz